### PR TITLE
[0.66-stable]Adding patches required for running RN66 in Office apps

### PR DESCRIPTION
**Pull request by mganandraj**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

## Summary

Patches already applied in 0.64 but weren't brought to main
New patches to bring new SO files to devmain

This change is cherry picked from main
`git cherry-pick cadd8890f761d4e2e4a331a0db0d4ce2c9c91d85`

## Changelog

Patches already applied in 0.64 but weren't brought to main
New patches to bring new SO files to devmain

This change is cherry picked from main
`git cherry-pick cadd8890f761d4e2e4a331a0db0d4ce2c9c91d85`

[CATEGORY] [TYPE] - Message

### CODE REVIEW

File: android-patches/patches/OfficeRNHost/ReactAndroid/src/main/java/com/facebook/react/bridge/ReactBridge.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
++    }catch (UnsatisfiedLinkError jscE){}
++
++    SoLoader.loadLibrary("glog");
++    SoLoader.loadLibrary("glog_init");
++    SoLoader.loadLibrary("fb");
++    SoLoader.loadLibrary("fbjni");
++    SoLoader.loadLibrary("yoga");
++    SoLoader.loadLibrary("folly_json");
++    SoLoader.loadLibrary("reactperfloggerjni");
++    SoLoader.loadLibrary("jsinspector");
```
Comment (rasaha91) [inline android-patches/patches/OfficeRNHost/ReactAndroid/src/main/java/com/facebook/react/bridge/ReactBridge.java:25]: does this need to be loaded in all scenarios? What about release builds?

File: android-patches/patches/JniUtils/ReactAndroid/src/main/java/com/facebook/react/fabric/jni/Android.mk [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
++LOCAL_SHARED_LIBRARIES := libjsi libreactconfig librrc_slider librrc_progressbar librrc_switch librrc_modal libyoga libglog libfb libfbjni libglog_init libfolly_json libfolly_futures libreact_render_mounting libreactnativejni libreact_utils libreact_render_debug libreact_render_graphics libreact_render_core react_render_componentregistry librrc_view librrc_unimplementedview librrc_root librrc_scrollview libbetter libreact_render_attributedstring libreact_render_uimanager libreact_render_templateprocessor libreact_render_scheduler libreact_render_animations libreact_render_imagemanager libreact_render_textlayoutmanager libreact_codegen_rncore rrc_text librrc_image librrc_textinput libreact_debug libreact_render_mapbuffer libmapbufferjni libreact_render_telemetry
```
Comment (rasaha91) [inline android-patches/patches/JniUtils/ReactAndroid/src/main/java/com/facebook/react/fabric/jni/Android.mk:8]: Is it possible to extend the macro definition across multiple lines (like how we do with `\` in nmake)? That way we could keep the android specific libraries on separate lines, which would help us stay in sync with any additions/removals that might happen in main or 0.66-stable. i.e. the patch shouldn't be manipulating the existing list, just adding new entries on new lines if possible

**Review by rasaha91 — APPROVED** [review]

File: android-patches/patches/OfficeRNHost/ReactAndroid/src/main/java/com/facebook/react/bridge/ReactBridge.java (hunk 1 above)
Comment (mganandraj) [reply android-patches/patches/OfficeRNHost/ReactAndroid/src/main/java/com/facebook/react/bridge/ReactBridge.java:25]: Yeah.. these dependencies comes from OSS builds .. and they are not flavor specific .. This are lots of room for improvement in the native builds in OSS ..

File: android-patches/patches/JniUtils/ReactAndroid/src/main/java/com/facebook/react/fabric/jni/Android.mk (hunk 2 above)
Comment (mganandraj) [reply android-patches/patches/JniUtils/ReactAndroid/src/main/java/com/facebook/react/fabric/jni/Android.mk:8]: Yeah .. good point ! Let's fix in main and cherry pick it to the stable branch ..

**Pull request merged**